### DISCUSSION TITLE
Dynamic routing works locally but hits 500 | Internal Server Error when hosted on Firebase.

### DISCUSSION THREAD

**Discussion by TCashion** [vercel/next.js#17498] [Help]
I have an app bootstrapped with: 

`npx create-next-app --example with-firebase-hosting`

And I set up a projects page directory like so: 

```
src
├── lib
│   └── projectsService.tsx.   <----- data is pulled from here
├── pages
│   ├── _app.js
│   ├── about.tsx
│   ├── admin
│   │   ├── index.tsx
│   │   └── login.tsx
│   ├── blog.tsx
│   ├── index.tsx
│   └── projects
│       ├── [id].tsx     <---- dynamically created pages causing error
│       └── index.tsx
```

Currently, the data coming into the dynamic pages (including the id for each page) is hard-coded into projectsService.tsx. The idea is to have this come from a database further down the line. 

In the local dev environment spun up using `npm run dev`, the dynamic routes to `/projects/*` work properly. However, when I deploy to Firebase, the routes give the standard NextJS `500 | internal server error` response. 

## Screenshots

Here is the network tab in the dev environment (properly working):
![Screen Shot 2020-09-30 at 2 29 59 PM](https://user-images.githubusercontent.com/58757351/94736428-6fd98180-0329-11eb-9192-1e462edd4a6a.png)

Here is the network tab in the production environment (error): 

![Screen Shot 2020-09-30 at 2 28 04 PM](https://user-images.githubusercontent.com/58757351/94736265-2ee16d00-0329-11eb-9b09-16bea93e16dd.png)

## Steps to Reproduce

Navigate to my [project](https://tcdotcom-migration.web.app/projects/) and, once there, click any button that says 'See More.'

From what I can tell it seems like I might need to change `next.config.js` to look for these specific routes, but I'm not exactly sure how to do that or if that's even the right place to look. 

Can anybody see what I'm doing wrong here? @jthegedus

**Comment by jthegedus** [comment]
Interesting, I have not come across this specific issue before. Did you make any progress on this you can share?

**Reply by almarrr** [reply]
I have the same problem with my catch all route. /[slug] works, but /[slug]/* gives the same error. It's from the root url, without /posts or anything.

**Reply by jthegedus** [reply]
When you hit a 500 server error, Next.js will output logs which can be viewed in the Cloud Function UI, can anyone share those here?

**Reply by rajeshwarsingh** [reply]
I am also facing the same issue, perhaps it's working in a dev environment.

**Reply by shiraazAh** [reply]
same issue happening for me

**Reply by bruceharrison1984** [reply]
Same issue here as well. Hitting the root URL works fine, but any other URL (`app/somePage`) fails with a 500

**Reply by moseslucas** [reply]
having the same issue with aws amplify next js 14

**Comment by TCashion** [comment]
Unfortunately I was never able to make progress on this. I removed my project from firebase and hosted through another service instead.

**Comment by nguyentuanit97** [comment]
I got the same error. Does anyone have an answer?

**Reply by vasu-analy** [reply]
I faced the same issue in netlify, but Its started working after upgraded to next@latest

**Comment by ekimcem** [comment]
Check your environment variables

**Comment by atco-jh** [comment]
Any solutions? Getting this issue too. I'm hosting SSG and SSR hybrid. SSG routes load fine, but every SSR route returns 500 error. No logs coming out. It works when I serve it locally using the emulator, but not when it's deployed. Nothing looks changed between the two environments. 

EDIT: Issue opened: #47642 and Resolved: #47699

**Comment by thiagomurtinho** [comment]
I recommend reading my response in another discussion. But it's probably related to your implementation of server-side functions.

[Error 500: Internal server error with getServerSideProps on production](https://github.com/vercel/next.js/discussions/20641#discussioncomment-5472761)

#20641
